### PR TITLE
[SIP-5] Refactor Time Series Table

### PR DESCRIPTION
**Pull request by kristw**
- Extract `slice` and `formData`
- Break rendering into smaller pieces for readability and ease of debugging. 
- Sort out what each row is (metric or column, object or string). Some variables renamed for clarity. 

Verify same slice on production and produce the same result.

### CODE REVIEW

File: superset/assets/src/visualizations/TimeTable/TimeTable.jsx [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+const defaultProps = {
+  className: '',
+  height: undefined,
+  url: '',
+};
+
+class TimeTable extends React.PureComponent {
+  renderLeftCell(row) {
+    const { rowType, url } = this.props;
+    // const context = { ...fd, metric };
```
Comment (williaster) [inline superset/assets/src/visualizations/TimeTable/TimeTable.jsx]: remove?

**Review by williaster — APPROVED** [review]
LGTM! 

This is so much more readable, thanks for taking the time for that!

File: superset/assets/src/visualizations/TimeTable/TimeTable.jsx (hunk 1 above)
Comment (kristw) [reply superset/assets/src/visualizations/TimeTable/TimeTable.jsx]: Does the url templating need anything other than the `metric`?
The commented was the original code and I was not sure if the url template wants something else from `formData`.

Comment (williaster) [reply superset/assets/src/visualizations/TimeTable/TimeTable.jsx]: Ohh I see. I think Michelle knows more about templating or @mistercrunch

Comment (kristw) [reply superset/assets/src/visualizations/TimeTable/TimeTable.jsx]: Hi @michellethomas , we would love your thoughts about the url template in time series table.

Comment (michellethomas) [reply superset/assets/src/visualizations/TimeTable/TimeTable.jsx]: I think it's was created to be used with metric, I can't think of anything else in formData that would be useful in the url. If you want to be sure, we can query charts that use this feature.

**Comment by williaster** [comment]
@kristw is this one ready to merge when the build passes?

**Comment by kaitorecca** [comment]
Then this one maybe the reason to break the Time Series Table features :(

**Pull request merged**